packages/worker/src/worker.ts

```diff
@@ -952,7 +957,27 @@ export class Worker<T extends WorkerSpec = DefaultWorkerSpec> {
           const activation = coresdk.workflow_activation.WFActivation.decode(new Uint8Array(buffer));
           const { runId, ...rest } = activation;
           this.log.debug('Got workflow activation', { runId, ...rest });
-          span.setAttribute('runId', runId).setAttribute('numJobs', rest.jobs.length);
+
+          let linkedContext = this.runIdsToSpanContext.get(runId);
+          span.setAttribute('run_id', runId).setAttribute('numJobs', rest.jobs.length);
+          // If there is an otel context in the headers, link our trace for the workflow to the
+          // trace in the header.
+          for (const j of rest.jobs) {
+            if (j.startWorkflow != null) {
+              const ctx = await extractSpanContextFromHeaders(j.startWorkflow.headers ?? {});
+              if (ctx != null && linkedContext == null) {
```

Comment:
Again these are `undefined`s not null.
I'll add a lint rule to prohibit using `==`.

Reply:
I wouldn't just b/c we still need it for not overly verbose checking of null/undef in protos